public/app/features/teams/TeamList.tsx

```diff
@@ -65,6 +65,10 @@ export class TeamList extends PureComponent<Props, State> {
     this.props.setSearchQuery(value);
   };
 
+  getRoleOptions = () => {
```

Comment:
This doesn't seem to be used anywhere? Also seems weird to wrap state in promise just to make a return value async.

Reply:
Yes i agree, and it was how it worked before. This was added by mistake. I added it to test if the issue was anonymous functions triggering all the extra api calls which it was